create-and-save-map/src/main/java/com/esri/arcgismaps/sample/createandsavemap/screens/MainScreen.kt

```diff
@@ -0,0 +1,439 @@
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.layers.Layer
+import com.arcgismaps.portal.PortalFolder
+import com.arcgismaps.toolkit.geoviewcompose.MapView
+import com.esri.arcgismaps.sample.createandsavemap.components.MapViewModel
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialog
+import com.esri.arcgismaps.sample.sampleslib.components.SampleTopAppBar
+
+/**
+ * Main screen layout for the sample app
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun MainScreen(sampleName: String) {
+    // create a ViewModel to handle MapView interactions
+    val mapViewModel: MapViewModel = viewModel()
+
+    val composableScope = rememberCoroutineScope()
+    val snackbarHostState = remember { SnackbarHostState() }
+    val controlsBottomSheetState = rememberModalBottomSheetState(skipPartiallyExpanded = false)
+    var isBottomSheetVisible by remember { mutableStateOf(false) }
+
+    Scaffold(
+        snackbarHost = { SnackbarHost(hostState = snackbarHostState) },
+        topBar = { SampleTopAppBar(title = sampleName) },
+        content = {
+            Box(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .padding(it),
+                contentAlignment = Alignment.Center
+            ) {
+                // map view not shown until the portal is loaded and a basemap has been set
+                MapView(
+                    arcGISMap = mapViewModel.arcGISMap,
+                    Modifier.fillMaxSize(),
+                )
+
+                // show the "Edit map" button only when the bottom sheet is not visible
+                if (!isBottomSheetVisible) {
+                    FloatingActionButton(
+                        modifier = Modifier
+                            .align(Alignment.BottomEnd)
+                            .padding(bottom = 36.dp, end = 24.dp),
+                        onClick = {
+                            isBottomSheetVisible = true
+                        })
+                    {
+                        Icon(Icons.Filled.Edit, contentDescription = "Edit map button")
+                        Spacer(Modifier.padding(8.dp))
+                    }
+                }
+
+                if (isBottomSheetVisible) {
+                    ModalBottomSheet(
+                        modifier = Modifier.wrapContentHeight(),
+                        sheetState = controlsBottomSheetState,
+                        onDismissRequest = {
+                            isBottomSheetVisible = false
+                        }
+                    ) {
+                        Column(
+                            Modifier
+                                .padding(12.dp)
+                                .navigationBarsPadding()
```

Comment:
Enable scrolling for this layout for landscape mode: 
```suggestion
                        Column(
                            Modifier
                                .padding(12.dp)
                                .navigationBarsPadding()
                                .verticalScroll(rememberScrollState())
```